### PR TITLE
bf: VLTCLT-2 set default region

### PR DESCRIPTION
**Pull request by naren-rajendran**
set default region to us-east-1 if one is not provided with the incoming request or if it is invalid.

### CODE REVIEW

**Comment by naren-scality** [comment]
@bert-e reset

File: lib/IAMClient.js [hunk 1]
```diff
@@ -391,16 +391,19 @@ class VaultClient {
             'signature is required');
         assert(typeof accessKey === 'string' && accessKey !== '',
             'accessKey is required');
-        assert(typeof region === 'string' && region !== '',
-            'region is required');
         assert(typeof scopeDate === 'string' && scopeDate !== '',
             'scopeDate is required');
+        // set region to 'us-east-1' if it is not provided
+        if (typeof region !== 'string' || region.trim() === '') {
+            // eslint-disable-next-line no-param-reassign
```
Comment (anurag4DSB) [inline lib/IAMClient.js:398]: I think, ideally we should respect this rule as that's the standard we follow.

**Review by anurag4DSB — COMMENTED** [review]
Vault client repo has just some basic tests. The effect of the change can be determined once we update the hash in vault and cloudserver repositories.

As this affects Auth, we cannot merge this, till we have a green build on cloudserver and vault with the updated hash of vaultclient.

**Review by nicolas2bert — APPROVED** [review]

**Review by alexanderchan-scality — APPROVED** [review]

**Pull request merged**